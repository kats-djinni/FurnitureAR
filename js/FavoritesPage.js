"use strict";

import React, { Component } from "react";
import {
  StyleSheet,
  View,
  Image,
  FlatList,
  TouchableHighlight,
  Dimensions
} from "react-native";
import { connect } from "react-redux"
import { Card, Text, Button } from "react-native-elements";
import { removeFavorite, getAllFavorites, removeAllFavorites } from "./store/favorites"
import FavoriteButton from "./FavoriteButton";

export class FavoritesPage extends Component {
  constructor() {
    super()
    this.state = {
      favorites: []
    }
  }

  async componentDidMount() {
    await this.props.getFavorites()
    .then(favorites => {
      this.setState({favorites: this.props.favorites})
    })
    .catch(console.error)
  }

  _deleteFavorite = async (index) => {
    await this.props.removeFavorite(index)
    await this.props.getFavorites()
    this.setState({
      favorites: this.props.favorites
    })
  }

  render() {
    return !this.props.favorites.length ? (
      <View style={styles.listContainer}>
        <Text style={styles.AllProductPage}>No favorites</Text>
      </View>
    ) : (
      <View style={styles.listContainer}>
        <Text h4 style={styles.header}>
          Favorites
        </Text>

        <FlatList
          data={this.state.favorites}
          renderItem={({ item, index }) => (
            <View>
              <Card>
                <Text style={styles.itemName}>{item.displayName}</Text>
                <Image
                  style={styles.faveImage}
                  source={{ uri: item.thumbnail }}
                />
              </Card>
              <TouchableHighlight onPress={() => this._deleteFavorite(index)}>
                <View style={styles.imageContainer}>
                  <Image
                    tintColor="red"
                    source={require("./res/icons/clear-icon.png")}
                  />
                </View>
              </TouchableHighlight>
            </View>
          )}
          keyExtractor={(item, index) => index.toString()}
        />
        <Button
          type="outline"
          raised={true}
          buttonStyle={styles.clearButton}
          title="Clear All"
          containerStyle={{ width: Dimensions.get("window").width * 0.55 }}
          onPress={() => this.props.clearFavorites()}
        />
      </View>
    );
  }
}

var styles = StyleSheet.create({
  listContainer: {
    flex: 1,
    flexDirection: "column",
    flexWrap: "wrap",
    fontFamily: "Arial",
    fontSize: 50,
    color: "#444B50",
    textAlignVertical: "center",
    textAlign: "center",
    margin: 20
  },
  SingleItem: {
    flex: 1
  },
  faveImage: {
    resizeMode: "cover",
    width: 150,
    height: 150
  },

  itemName: {
    fontWeight: "bold",
    textAlign: "center",
    padding: 5
  },

  imageContainer: {
    justifyContent: "center",
    alignItems: "center"
  },

  header: {
    textAlign: "center",
    color: "#444B50"
  },

  clearButton: {
    // width: Dimensions.get("window").width * 0.5
  }
})

const mapStateToProps = (state) => {
    return {
      favorites: state.favorites.favorites
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getFavorites: () => dispatch(getAllFavorites()),
    removeFavorite: (index) => dispatch(removeFavorite(index)),
    clearFavorites: () => dispatch(removeAllFavorites())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(FavoritesPage)
