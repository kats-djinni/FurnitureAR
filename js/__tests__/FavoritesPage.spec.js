import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { FavoritesPage } from '../FavoritesPage'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
enzyme.configure({ adapter: new Adapter() })
import { FlatList } from 'react-native'


describe('<FavoritesPage /> component', () => {

    const faves = [{
                "name": "assets/3txPAhYeu-x",
                "displayName": "Chair",
                "authorName": "Ryan D",
                "objurl": "https://poly.googleapis.com/downloads/fp/1551711596400293/3txPAhYeu-x/3DjXPukLzqV/model.obj",
                "mtlurl": "https://poly.googleapis.com/downloads/fp/1551711596400293/3txPAhYeu-x/3DjXPukLzqV/materials.mtl",
                "formatType": "OBJ",
                "thumbnail": "https://lh3.googleusercontent.com/dlwexhUXTVDfcmJayQRruP6IqK35UG1hw5C8Wmv-5E7d-cAfVQf5ThJ1cmiayHc"
            }
        ]

    let wrapper
    let wrapperInstance

    beforeEach(() => {
       const wrapper = shallow(<FavoritesPage />)
       //wrapperInstance = wrapper.instance() 

       wrapper.setState({ favorites: faves}, () => {
           wrapper.update();
           expect(wrapper.state('favorites')).toEqual(faves);
           //console.log(wrapper.debug());
       });

    })
 
    it('renders an FlatList on Favorites', () => {
        const wrapper = shallow(<FavoritesPage />)
        wrapper.setState({ favorites: faves}, () => {
           wrapper.update();
           expect(wrapper.find(FlatList).length).toEqual(1)
           //console.log(wrapper.debug());
       });
        
      })

    it('renders correctly', () => {
        const wrapper = shallow(<FavoritesPage />)
        expect(wrapper).toMatchSnapshot()
        })
})
