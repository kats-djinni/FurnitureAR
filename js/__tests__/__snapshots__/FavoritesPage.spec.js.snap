// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesPage /> component renders correctly 1`] = `ShallowWrapper {}`;
